src/multiselect.js

```diff
@@ -0,0 +1,206 @@
+/**
+ * @license
+ * Copyright 2022 MIT
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Plugin main class.
+ */
+
+import * as Blockly from 'blockly/core';
+
+import * as ContextMenu from './multiselect_contextmenu';
+import * as Shortcut from './multiselect_shortcut';
+import {blockSelection, inMultipleSelectionMode,
+  hasSelectedParent, setBaseBlockDragger} from './global';
+import {MultiselectControls} from './multiselect_controls';
+
+/**
+ * Class for using multiple select blocks on workspace.
+ */
+export class Multiselect {
+  /**
+   * Initalize the class data structure.
+   * @param {!Blockly.WorkspaceSvg} workspace The workspace to sit in.
+   */
+  constructor(workspace) {
+    this.workspace_ = workspace;
+    this.origHandleWsStart_ = Blockly.Gesture.prototype.handleWsStart;
+  }
+
+  /**
+   * Bind the events and replace registration.
+   * @param {{multiselectIcon: Object, useDoubleClick: boolean}} options
+   * to set.
+   */
+  init(options) {
+    this.onKeyDownWrapper_ = Blockly.browserEvents.conditionalBind(
+        this.workspace_.getInjectionDiv(), 'keydown', this, this.onKeyDown_);
+    this.onKeyUpWrapper_ = Blockly.browserEvents.conditionalBind(
+        this.workspace_.getInjectionDiv(), 'keyup', this, this.onKeyUp_);
+    this.eventListenerWrapper_ = this.eventListener_.bind(this);
+    this.workspace_.addChangeListener(this.eventListenerWrapper_);
+
+    ContextMenu.unregisterContextMenu();
+    ContextMenu.registerOurContextMenu();
+    Shortcut.unregisterShortcut();
+    Shortcut.registerOurShortcut();
+
+    this.controls_ = new MultiselectControls(
+        this.workspace_, options.multiselectIcon, this);
```

Comment:
Do we want to make it so that the position of the control can be customized? Where does it end up by default? Is this documented somewhere?

Reply:
It's always located near the trashcan. You can check the location in different modes.
https://github.com/mit-cml/workspace-multiselect/pull/1#discussion_r921757035

I guess it may not be necessary to make position of the control customizable as I can see that trashcan also has a fixed position.

https://developers.google.com/blockly/guides/configure/web/configuration_struct
<html>
<body>
<!--StartFragment-->

trashcan | boolean | Displays or hides the trashcan. Defaults to true if the toolbox has categories, false otherwise.
-- | -- | --


<!--EndFragment-->
</body>
</html>